components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/util/APIFileUtils.java

```diff
@@ -350,6 +361,90 @@ public static InputStream readFileContentAsStream(String path) throws APIMgtDAOE
     }
 
     /**
+     * Creates a zip archive from the given {@link InputStream} inputStream
+     *
+     * @param inputStream {@link InputStream} instance
+     * @param archivePath path to create the zip archive
+     * @throws APIMgtDAOException if an error occurs while creating the archive
+     */
+    public static void createArchiveFromInputStream(InputStream inputStream, String archivePath)
+            throws APIMgtDAOException {
+
+        FileOutputStream outFileStream = null;
+        try {
+            outFileStream = new FileOutputStream(new File(archivePath));
+            IOUtils.copy(inputStream, outFileStream);
+        } catch (IOException e) {
+            throw new APIMgtDAOException("Error in Creating archive from uploaded data", e);
+        } finally {
```

Comment:
Its nicer to use 'try with resources' for the FileOutputStream. Then we don't need have a finally block for closing